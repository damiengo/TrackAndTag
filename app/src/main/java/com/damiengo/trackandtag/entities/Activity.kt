package com.damiengo.trackandtag.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity
data class Activity(
    @PrimaryKey val activityId: Long,
    @ColumnInfo(name = "title") val title: String?,
    @ColumnInfo(name = "description") val description: String?,
    @ColumnInfo(name = "number") val number: Float?,
    @ColumnInfo(name = "made_at") val madeAt: LocalDateTime?,
    @ColumnInfo(name = "created_at") val createdAt: LocalDateTime?,
    @ColumnInfo(name = "updated_at") val updatedAt: LocalDateTime?
)
