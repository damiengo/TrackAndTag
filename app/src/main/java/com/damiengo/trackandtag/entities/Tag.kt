package com.damiengo.trackandtag.entities

import android.os.Build
import androidx.annotation.RequiresApi
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity
data class Tag(
    @PrimaryKey(autoGenerate = true) var tagId: Long,
    @ColumnInfo(name = "text") val title: String?,
    @ColumnInfo(name = "created_at") val createdAt: LocalDateTime?,
    @ColumnInfo(name = "updated_at") val updatedAt: LocalDateTime?
) {
    @RequiresApi(Build.VERSION_CODES.O)
    @Ignore
    constructor(text: String = "",
                createdAt: LocalDateTime? = LocalDateTime.now()) : this(0, text, createdAt, null)
}
