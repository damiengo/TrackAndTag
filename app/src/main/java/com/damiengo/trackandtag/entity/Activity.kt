package com.damiengo.trackandtag.entity

import android.os.Build
import androidx.annotation.RequiresApi
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity
data class Activity(
    @PrimaryKey(autoGenerate = true) var activityId: Long,
    @ColumnInfo val title: String?,
    @ColumnInfo val description: String?,
    @ColumnInfo val number: Float?,
    @ColumnInfo val madeAt: Long,
    @ColumnInfo val createdAt: Long,
    @ColumnInfo val updatedAt: Long
) {
    @RequiresApi(Build.VERSION_CODES.O)
    @Ignore
    constructor(title: String = "",
                description: String = "",
                number: Float? = 0f,
                madeAt: Long = System.currentTimeMillis(),
                createdAt: Long = System.currentTimeMillis()) : this(0, title, description, number, madeAt, createdAt, 0)
}