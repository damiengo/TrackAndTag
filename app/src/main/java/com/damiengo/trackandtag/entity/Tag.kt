package com.damiengo.trackandtag.entity

import android.os.Build
import androidx.annotation.RequiresApi
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity
data class Tag(
    @PrimaryKey(autoGenerate = true) var tagId: Long,
    @ColumnInfo val text: String?,
    @ColumnInfo val createdAt: Long,
    @ColumnInfo val updatedAt: Long
) {
    @RequiresApi(Build.VERSION_CODES.O)
    @Ignore
    constructor(text: String = "",
                createdAt: Long = System.currentTimeMillis()) : this(0, text, createdAt, 0)
}
