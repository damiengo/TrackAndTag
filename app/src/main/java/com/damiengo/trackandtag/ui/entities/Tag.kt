package com.damiengo.trackandtag.ui.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity
data class Tag(
    @PrimaryKey val uid: Int,
    @ColumnInfo(name = "text") val title: String?,
    @ColumnInfo(name = "created_at") val createdAt: LocalDateTime?,
    @ColumnInfo(name = "updated_at") val updatedAt: LocalDateTime?
)
