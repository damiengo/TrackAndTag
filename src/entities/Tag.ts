import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  ManyToMany, 
  JoinTable,
} from 'typeorm';

import { Activity } from './Activity'

@Entity('tag')
export class Tag {

  @PrimaryGeneratedColumn() id: number;
  @Column('text') label: string;
  @Column('text', { nullable: true }) description: string | null;
  @Column({ nullable: true }) unit: number | undefined;
  @Column() createdAt: number;
  @Column() updatedAt: number;

  @ManyToMany(() => Activity, activity => activity.tags)
  @JoinTable()
  activities: Activity[];

}