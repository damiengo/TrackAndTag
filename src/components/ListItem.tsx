import React from 'react'
import {
  View,
  Text, 
  TouchableWithoutFeedback
} from 'react-native';
import Moment from 'moment';

import { Tag } from '../entities/Tag'

export default function ListItem({ navigation, activity }: any) {

    const genTags = (tags: Tag[]): string => {
        return tags.map((tag) => {
            return '#'+tag.label
        }).join(' â€¢ ')
    }

    return (
        <TouchableWithoutFeedback onPress={() => navigation.navigate('Edit', {item: activity})}>
          <View
            style={{
              flexDirection: 'row',
              flex: 1,
              marginTop: 5,
              alignItems: 'center'
            }}
          >
            <Text style={{marginRight: 5, marginLeft: 5}}>{Moment(activity.made_at).format('DD/MM')}</Text> 
            <Text style={{marginRight: 5, paddingLeft: 4, paddingRight: 4, backgroundColor: '#55FF00'}}>{activity.quantity}</Text> 
            <Text style={{marginRight: 5, fontWeight: 'bold'}}>{activity.title}</Text> 
            <Text style={{marginRight: 5, fontSize: 13}} testID='input_tag'>{genTags(activity.tags)}</Text> 
          </View>
        </TouchableWithoutFeedback>
    )

}