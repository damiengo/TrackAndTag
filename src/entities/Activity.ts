import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  ManyToMany, 
  JoinTable,
} from 'typeorm';

import { Tag } from './Tag'

@Entity('activity')
export class Activity {

  @PrimaryGeneratedColumn() id: number;
  @Column('text') title: string;
  @Column('text', { nullable: true }) description: string | null;
  @Column() quantity: number;
  @Column() madeAt: number;
  @Column() createdAt: number;
  @Column({ nullable: true }) updatedAt: number | null;

  @ManyToMany(() => Tag, (tag) => tag.activities)
  @JoinTable()
  tags: Tag[];

}