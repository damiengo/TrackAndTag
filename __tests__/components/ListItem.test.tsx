import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';

import ListItem from '../../src/components/ListItem';
import { Activity } from '../../src/entities/Activity';
import { Tag } from '../../src/entities/Tag';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('add an item', () => {
    const navigation = {
        navigate: jest.fn()
    }
    const tag1 = new Tag()
    tag1.id = 1
    tag1.label = 'tag 1'
    const tag2 = new Tag()
    tag2.id = 2
    tag2.label = 'tag 2'

    const activity = new Activity();
    activity.id = 1
    activity.title = 'The title'
    activity.description = 'A short description'
    activity.quantity = 18
    activity.madeAt = (new Date()).getTime()
    activity.tags = [tag1, tag2]

    const wrapper = shallow(<ListItem navigation={navigation} activity={activity} />)
    
    wrapper.find(TouchableWithoutFeedback).simulate('press')
    expect(navigation.navigate.mock.calls.length).toEqual(1)
    expect(navigation.navigate).toBeCalledWith('Edit', { item: activity })
    expect(wrapper).toMatchSnapshot()
});