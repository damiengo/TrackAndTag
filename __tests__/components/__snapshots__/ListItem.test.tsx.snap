// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add an item 1`] = `ShallowWrapper {}`;
